Remove shutdown event trigger

PeterSuh-Q3 opened the pull request:
It shouldn't be repeated every time you shutdown.

I checked the contents of your apl-reboot.sh .
This script should not be run automatically by the scheduler whenever a shutdown event occurs.

https://github.com/wjz304/arpl-i18n/blob/main/files/board/arpl/overlayfs/usr/sbin/arpl-reboot.sh

By the way, has this arpl-reboot.sh ever been tested?
The path of this file is /usr/sbin/arpl-reboot.sh,
In the install.sh script
/usr/bin/arpl-reboot.sh
"INSERT INTO task VALUES('RebootToArpl', '', '-', '', 0, 0, 0, 0, '', 0, '/usr/bin/arpl-reboot.sh "config"', 'script', '{}', '', '', '{}', '{}');"
path is not correct.
Strange.

PeterSuh-Q3 commented:
[esynoscheduler.db.zip](https://github.com/wjz304/arpl-addons/files/11580530/esynoscheduler.db.zip)

The attached file is from esynoscheduler.db
The shutdown event has been removed.

wjz304 commented:
![image](https://github.com/wjz304/arpl-addons/assets/5615843/2367c1f5-71fd-45e8-bea9-86f91b72516d)
Yes, he doesn't run every time he shuts down. The created task is not enabled, and when you need to enter the buildroot, you need to manually run it on the task interface. The event key seems to be a mandatory option, and you can only choose to shutdown or startup.

In buildroot, arpl reboot. sh is in/usr/sbin/,
In DSM, arpl reboot. sh is in/usr/bin/,   (misc/install.sh  L:9)

PeterSuh-Q3 commented:
When creating an event key in the control panel, something must be specified as a required value.
However, the scheduler generated by the modified install.sh script as above
next-runtime is removed and
Events are also removed with "-".

![스크린샷 2023-05-27 오전 10 17 48](https://github.com/wjz304/arpl-addons/assets/85427533/1e525649-4f98-4b76-8420-391fe2363e5b)

![스크린샷 2023-05-27 오전 10 19 39](https://github.com/wjz304/arpl-addons/assets/85427533/58930380-b78e-4aeb-84f9-d878168dea71)

It's kind of strange that the shutdown event doesn't work every time even though it's logged.
As a result of my testing, the event was triggered every time I shut down the system.

wjz304 commented:
If the task is in a disable state, I don't think it will be called. 

i think you can delete it and test it? If you delete it and still called, the reboot.sh script maybe called elsewhere?

PeterSuh-Q3 commented:
Yes I already deleted shuwdown and tested it.
Deleted results will not automatically trigger an event.
It only works if I run it manually.
Wouldn't this be the right way?

![스크린샷 2023-05-27 오전 10 47 03](https://github.com/wjz304/arpl-addons/assets/85427533/1b7f5740-bada-4905-b332-5bf959f2b3b6)
![스크린샷 2023-05-27 오전 10 47 28](https://github.com/wjz304/arpl-addons/assets/85427533/27cce851-ca26-46d8-80b9-60e462441d72)

wjz304 commented:
> Yes I already deleted shuwdown and tested it. Deleted results will not automatically trigger an event. It only works if I run it manually. Wouldn't this be the right way?
> 
> ![스크린샷 2023-05-27 오전 10 47 03](https://user-images.githubusercontent.com/85427533/241348140-1b7f5740-bada-4905-b332-5bf959f2b3b6.png) ![스크린샷 2023-05-27 오전 10 47 28](https://user-images.githubusercontent.com/85427533/241348142-27cce851-ca26-46d8-80b9-60e462441d72.png)

That's right, it should be like this.

wjz304 commented:
```
# CREATE TABLE if not exists task (
#                         task_name                       TEXT PRIMARY KEY NOT NULL,
#                         description                     TEXT,
#                         event                           TEXT NOT NULL,
#                         depend_on_task          TEXT,
#                         enable                          INTEGER DEFAULT 1,
#                         owner                           INTEGER NOT NULL,
#                         run_the_same_time       INTEGER DEFAULT 1,
#                         notify_enable           INTEGER DEFAULT 0,
#                         notify_mail                     TEXT,
#                         notify_if_error         INTEGER DEFAULT 0,
#                         operation                       TEXT NOT NULL,
#                         operation_type          TEXT NOT NULL,
#                         status                          TEXT NOT NULL DEFAULT "{}",
#                         last_start_time         INTEGER,
#                         last_stop_time          INTEGER,
#                         last_exit_info          TEXT,
#                         extra                           TEXT DEFAULT "{}"
# );
```
According to the table structure, the event key is sufficient for non empty text,

But I don't recommend changing it to the option provided on non UI, because we don't know the rules when Syno's application reading the db.

PeterSuh-Q3 commented:
I have nothing to say if you're following the principles in XPE.
And I obviously didn't replace shutdown with NULL
I was instructed to fill it with "-".
I only talk about the results after going through all the tests.
Try it the way you want.
You can ignore this request if it is not necessary.

wjz304 commented:
I saw '-',

Although it is defined as non empty text in the database, it only has two options on the UI,

Therefore, the approximate judgment of the syno system is "if else", 
If it is other, it is highly likely to still be classified as one of them.(of course, it's just a guess)

According to my test, manually running the "event" key will not affect the execution of the task. If there is an impact, it may not be the root cause.

PeterSuh-Q3 commented:
I will show that it is possible to create a scheduled task that does not specify a trigger event. (It is not a triggered schedule.)

![스크린샷 2023-05-27 오전 11 48 21](https://github.com/wjz304/arpl-addons/assets/85427533/363cb7a8-4aa5-422c-b867-6c6ebed256df)

![스크린샷 2023-05-27 오전 11 42 01](https://github.com/wjz304/arpl-addons/assets/85427533/bd199b41-77a1-4907-80ef-5f57f1fb82ed)

![스크린샷 2023-05-27 오전 11 42 05](https://github.com/wjz304/arpl-addons/assets/85427533/1e8b3e3d-84cd-4b28-91eb-3fcf774dc19f)

![스크린샷 2023-05-27 오전 11 42 21](https://github.com/wjz304/arpl-addons/assets/85427533/1092d380-9547-4d0f-a306-3403828c1643)

If arpl insists that the shutdown event never triggers even if it is specified, I won't argue any further.
However, in tcrp it continues to become a tree and the grub menu loops in an undesirable direction.

wjz304 commented:
https://raw.githubusercontent.com/PeterSuh-Q3/tcrp-addons/main/reboottotcrp/src/tcrp-reboot.sh#:~:text=i%20%22/set%20default%3D%5C%22-,*,-%5C%22/cset%20default%3D%5C%223

`sed -i "/^set default=\"[0-9]\+\"/cset default=\"3\"" grub.cfg`
I suggest modifying this place, it will also replace set default="${saved_entry}"

PeterSuh-Q3 commented:
https://github.com/PeterSuh-Q3/tcrpfriend/blob/main/buildroot/board/tcrpfriend/rootfs-overlay/root/boot.sh#L311

Already in FRIEND's boot.sh there was an example where your advice was used.
thank you I'll fix it according to your advice.
